source/common/regular-expressions.js

```diff
@@ -57,25 +353,71 @@ module.exports = {
       'i' + flag
     )
   },
+
   /**
-   * Returns a regular expression that matches file IDs as in the settings
-   *
-   * @param   {boolean}  multiline  Whether to match multiline
+   * Returns a regular expression that matches MarkDown Tables.
    *
    * @return  {RegExp}              The compiled Regular Expression
    */
-  'getIDRE': function (multiline = false) {
-    let flag = (multiline) ? 'm' : ''
-    let idRegExpString = global.config.get('zkn.idRE')
-    // Make sure the ID definitely has at least one
-    // capturing group to not produce errors.
-    if (!(/\(.+?\)/.test(idRegExpString))) {
-      idRegExpString = `(${idRegExpString})`
-    }
+  'getTableRE': function () {
+    return RegExp(
+      /^\|.+\|$/.source,
+      'i')
+  },
 
+  /**
+   * Returns a regular expression that matches inline frames.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getTableHeadingRE': function () {
     return RegExp(
-      idRegExpString,
-      'g' + flag
+      /(^[- ]+$)|(^[- +:]+$)|(^[- |:+]+$)/.source
     )
+  },
+
+  /**
+   * Returns a regular expression that matches tasks.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getTaskRE': function () {
+    return RegExp(
+      /^(\s*)([-+*]) \[( |x)\] /.source,
+      'g')
+  },
+
+  /**
+   * Returns a regular expression that matches URLs.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getUrlRE': function () {
+    return RegExp(
+      /^\[([^\]]+)\]\((.+?)\)|(((?:(?:aaas?|about|acap|adiumxtra|af[ps]|aim|apt|attachment|aw|beshare|bitcoin|bolo|callto|cap|chrome(?:-extension)?|cid|coap|com-eventbrite-attendee|content|crid|cvs|data|dav|dict|dlna-(?:playcontainer|playsingle)|dns|doi|dtn|dvb|ed2k|facetime|feed|file|finger|fish|ftp|geo|gg|git|gizmoproject|go|gopher|gtalk|h323|hcp|https?|iax|icap|icon|im|imap|info|ipn|ipp|irc[6s]?|iris(?:\.beep|\.lwz|\.xpc|\.xpcs)?|itms|jar|javascript|jms|keyparc|lastfm|ldaps?|magnet|mailto|maps|market|message|mid|mms|ms-help|msnim|msrps?|mtqp|mumble|mupdate|mvn|news|nfs|nih?|nntp|notes|oid|opaquelocktoken|palm|paparazzi|platform|pop|pres|proxy|psyc|query|res(?:ource)?|rmi|rsync|rtmp|rtsp|secondlife|service|session|sftp|sgn|shttp|sieve|sips?|skype|sm[bs]|snmp|soap\.beeps?|soldat|spotify|ssh|steam|svn|tag|teamspeak|tel(?:net)?|tftp|things|thismessage|tip|tn3270|tv|udp|unreal|urn|ut2004|vemmi|ventrilo|view-source|webcal|wss?|wtai|wyciwyg|xcon(?:-userid)?|xfire|xmlrpc\.beeps?|xmpp|xri|ymsgr|z39\.50[rs]?):(?:\/{1,3}|[a-z0-9%])|www\d{0,3}[.]|[a-z0-9.-]+[.][a-z]{2,4}\/)(?:[^\s()<>]|\([^\s()<>]*\))+(?:\([^\s()<>]*\)|[^\s`*!()[\]{};:'".,<>?«»“”‘’])))|([a-z0-9.\-_+]+?@[a-z0-9.\-_+]+\.[a-z]{2,7})$/.source,
```

Comment:
Note to myself: We definitely should get to simplifying this monster 👀

Reply:
I like needing to scroll for five minutes to parse a regex.